Sources/AsyncHTTPClient/ConnectionPool/HTTPConnectionPool+HTTP1State.swift

```diff
@@ -0,0 +1,694 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the AsyncHTTPClient open source project
+//
+// Copyright (c) 2021 Apple Inc. and the AsyncHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of AsyncHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+extension HTTPConnectionPool {
+    struct HTTP1ConnectionState {
+        enum State {
+            case starting(Waiter?)
+            case active(Connection, isAvailable: Bool, lastReturn: NIODeadline)
```

Comment:
What's `lastReturn`, the point in time the connection was returned to the pool?

Reply:
Is `lastReturn` ever relevant when `isAvailable` is `false`?

Reply:
I'm wondering if this is better expressed as

```swift
enum Availability {
  case available(since: NIODeadline)
  case unavailable
}

enum State {
  case active(Connection, Availability)
  // ...
}
```

The `availableAndLastReturn` could then be `availableSince` (I found the use of "return" in the various names in `executeRequest` quite unclear)

Reply:
I changed it to this, which I like even better:

```swift
        enum State {
            case starting(Waiter?)
            case available(Connection, since: NIODeadline)
            case leased(Connection)
            case failed
            case closed
        }
```